pulsar-function-go/pf/context.go

```diff
@@ -0,0 +1,98 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package pf
+
+import (
+	"context"
+)
+
+type FunctionContext struct {
+	InstanceConf *InstanceConf
+	UserConfigs  map[string]interface{}
+	InputTopics  []string
+}
+
+func NewFuncContext() *FunctionContext {
+	fc := &FunctionContext{
+		InstanceConf: NewInstanceConf(),
+		UserConfigs:  make(map[string]interface{}),
+	}
+	return fc
+}
+
+func (c *FunctionContext) GetInstanceID() int {
+	return c.InstanceConf.InstanceID
+}
+
+func (c *FunctionContext) GetInputTopics() []string {
```

Comment:
this should be changed to getting topics from functiondetails(like the java one)

Reply:
yes, `InputTopics` are passed in from functiondetails, as follows:

```
for topic, consumerConf := range funcDetails.Source.InputSpecs{
    ...
    gi.context.InputTopics = append(gi.context.InputTopics, topic)
}
```